### PR TITLE
feat: Added precommit support

### PR DESCRIPTION
**Pull request by CamPen21**
# Motivation

Suggested feature to be supported on issue #13.

# Changes

- Added .pre-commit-config.yaml

### trailing-whitespace

Adds a trailing whitespace at the end of every staged file. (Good practice IMO).

### end-of-file-fixer

Checks and cleans special Unicode codes that are invisible and can mess with the file (Copy and pasted code).

### check-yaml

Verifies general purpose yaml files.

### check-added-large-files

Safeguard from adding big stuff to the repo.

### CODE REVIEW

**Comment by CamPen21** [comment]
@Melkeydev I didn't know what other hooks you wanted past go linting and formatting so I added some pretty standard ones.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+        - id: trailing-whitespace
+        - id: end-of-file-fixer
+        - id: check-yaml
```
Comment (Melkeydev) [inline .pre-commit-config.yaml:9]: I dont think we need this hook right now

**Review by Melkeydev — CHANGES_REQUESTED** [review]
Small nit

**Review by Melkeydev — APPROVED** [review]
LTGM

**Pull request merged**